### PR TITLE
Support Kotlin 1.9.0

### LINKED ISSUE #379: Support Kotlin 1.8.21
Hello 👋 

Awesome work on the library! By any chance there will be a new version released soon that supports Kotlin v1.8.21? 

I tried updating it ([link](https://github.com/dariuszkuc/kotlin-compile-testing/commit/c6f8742aa4c8a3c79b6b5f32eaf0a1185efdb855)) but sadly there is some change in behavior in v1.8.20 that breaks JS compiler. Nothing jumps out from the [release notes](https://kotlinlang.org/docs/whatsnew1820.html) - they did add alpha support for new JS IR compiler but that "shouldn't" affect the legacy compiler.

After updating to v1.8.20+ JS compilation fails asking to set some IR properties (`-ir-output-dir` and `-ir-output-name`). I tried updating the default [`KotlinJsCompilation#jsArgs()`](https://github.com/tschuchortdev/kotlin-compile-testing/blob/master/core/src/main/kotlin/com/tschuchort/compiletesting/KotlinJsCompilation.kt#L75-L95) args to include those but that didn't work either, i.e. I added the following

```kotlin
args.moduleName = "KCT"
args.outputDir = outputDir.toString()
```

With the above JS compilation runs but does not produce any outputs.... Any help would be appreciated!

### PR DESCRIPTION
**Pull request by BoD**
A follow-up to #380 (and closes #379). 

Let's see if I have better luck with Windows tests.

### CODE REVIEW

File: core/src/main/kotlin/com/tschuchort/compiletesting/AbstractKotlinCompilation.kt [hunk 1]
```diff
@@ -36,8 +36,9 @@ abstract class AbstractKotlinCompilation<A : CommonCompilerArguments> internal c
     /** Working directory for the compilation */
     var workingDir: File by default {
         val path = Files.createTempDirectory("Kotlin-Compilation")
-        log("Created temporary working directory at ${path.toAbsolutePath()}")
-        return@default path.toFile()
+        val canonicalFile = path.toFile().canonicalFile
```
Comment (BoD) [inline core/src/main/kotlin/com/tschuchort/compiletesting/AbstractKotlinCompilation.kt:39]: The JS compiler calls .canonicalFile internally - at least on MacOS that may be a different to the absolute path (`/private/var/…` vs `/var/…`), which made the tests fail.

File: core/src/main/kotlin/com/tschuchort/compiletesting/KotlinJsCompilation.kt [hunk 2]
```diff
@@ -91,7 +95,7 @@ class KotlinJsCompilation : AbstractKotlinCompilation<K2JSCompilerArguments>() {
     args.irOnly = irOnly
     args.irModuleName = irModuleName
     args.generateDts = generateDts
-    args.useDeprecatedLegacyCompiler = useDeprecatedLegacyCompiler
+    args.forceDeprecatedLegacyCompilerUsage = useDeprecatedLegacyCompiler
```
Comment (BoD) [inline core/src/main/kotlin/com/tschuchort/compiletesting/KotlinJsCompilation.kt:98]: `useDeprecatedLegacyCompiler` was removed [here](https://github.com/JetBrains/kotlin/commit/4da81b2b6ea977204463b5c4c4da1d7c8d61314e#diff-24ba9c8290d5b9807cc16df8bad3dccab72ffcde4d87aab67a343dcff812f054L593)

File: core/src/test/kotlin/com/tschuchort/compiletesting/KotlinJsCompilationTests.kt [hunk 3]
```diff
@@ -78,27 +78,6 @@ class KotlinJsCompilationTests {
 		assertThat(jsFile.readText()).contains("function KSource_0() {")
 	}
 
-	@Test
```
Comment (BoD) [inline core/src/test/kotlin/com/tschuchort/compiletesting/KotlinJsCompilationTests.kt:81]: Unfortunately I’m not sure why this no longer works, it fails with `kotlinx.browser` not being resolved. Maybe this is no longer supported? But [the KDoc](https://kotlinlang.org/api/latest/jvm/stdlib/kotlinx.browser/) seems to indicate it should still work 🤔 The source is still [there](https://github.com/JetBrains/kotlin/blob/master/libraries/stdlib/js/src/kotlinx/browser/declarations.kt). Any idea?

**Comment by apatrida** [comment]
In this PR can we remove the call in `KotlinCompilation` for setting `useIR` as it is defaulted to `true` and is removed by Kotlin `1.9.20-Beta2` so this would just be another breaking change coming soon. 

Also this deprecation warning should be dealt with:
 https://youtrack.jetbrains.com/issue/KT-52665

File: core/src/main/kotlin/com/tschuchort/compiletesting/KotlinCompilation.kt [hunk 4]
```diff
@@ -71,6 +71,7 @@ class KotlinCompilation : AbstractKotlinCompilation<K2JVMCompilerArguments>() {
 	var javaParameters: Boolean = false
 
 	/** Use the IR backend */
+	@Deprecated("Since Kotlin 1.9.20 this option is no longer supported by the compiler. It has no effect.")
```
Comment (BoD) [inline core/src/main/kotlin/com/tschuchort/compiletesting/KotlinCompilation.kt:74]: Removed in [this commit](https://github.com/JetBrains/kotlin/commit/60016d3e5bb7c58d54bff3c082edfd53d0086218).

**Pull request merged**